pywr/core.py

```diff
@@ -1155,6 +1155,124 @@ def after(self, timestep):
+        of total flow to pass through the additional sublinks. If no ratio is required for a particular
+        sublink then use `None` for its items. The sum of the ratios must be <= 1.0.
+
+    Notes
+    -----
+    Users must be careful when using the ratio fixing mechanism. Ratios use the last non-split
+     sublink (i.e. X1 but not X0). If this link is constrained with a maximum or minimum flow,
+     or if it there is another unconstrained link (i.e. if X0 is unconstrained) then ratios
+     across this whole node may not be enforced as expected.
+
+    """
+    def __init__(self, *args, **kwargs):
+        self.allow_isolated = True
+        costs = list(kwargs.pop('cost'))
+        max_flows = list(kwargs.pop('max_flow'))
+
+        extra_slots = kwargs.pop('extra_slots', 1)
+        if extra_slots < 1:
+            raise ValueError("extra_slots must be at least 1.")
+
+        # No cost or maximum flow on the additional links
+        # The max_flows could be problematic with the aggregated node.
+        costs.extend([0.0]*extra_slots)
+        max_flows.extend([None]*extra_slots)
+        # Edit the kwargs to get the PiecewiseLink to setup as we want.
+        kwargs['cost'] = costs
+        kwargs['max_flow'] = max_flows
+
+        # Default to integer names
+        self.slot_names = list(kwargs.pop('slot_names', range(extra_slots+1)))
+        if extra_slots+1 != len(self.slot_names):
+            raise ValueError("slot_names must be one more than the number of extra_slots.")
+
+        ratios = kwargs.pop('ratios', None)
+        # Finally initialise the parent.
+        super(MultiSplitLink, self).__init__(*args, **kwargs)
+
+        self._extra_inputs = []
+        self._extra_outputs = []
+        n = len(self.sublinks) - extra_slots
+        for i in range(extra_slots):
+            # create a new input inside the piecewise link which only has access
+            # to flow travelling via the last sublink (X2)
+            otpt = Output(self.model, '{} Extra Output {}'.format(self.name, i), domain=self.sub_domain)
+            inpt = Input(self.model, '{} Extra Input {}'.format(self.name, i))
+
+            otpt.connect(inpt)
+            self.sublinks[n+i].connect(otpt)
+
+            self._extra_inputs.append(inpt)
+            self._extra_outputs.append(otpt)
+
+        # Now create an aggregated node for addition constaints if required.
+        if ratios is not None:
+            if extra_slots != len(ratios):
+                raise ValueError("ratios must have a length equal to extra_slots.")
+
+            nodes = []
+            valid_ratios = []
+            for r, nd in zip(ratios, self.sublinks[n:]):
+                if r is not None:
+                    nodes.append(nd)
+                    valid_ratios.append(r)
+
+            remaining_ratio = 1.0 - sum(valid_ratios)
+            if remaining_ratio < 0.0:
+                raise ValueError('Sum of ratios is more than 1.0.')
```

Comment:
Could we normalise the ratios instead? Some ratios aren't very nice to specify like this, e.g. 1/3rd would have to be `(0.3333333333, 0.666666666)`. Obviously in Python you could just pass `(1/3, 2/3)`, but this isn't possible in JSON. Instead it would be nicer to pass `(1, 2)`.